### PR TITLE
Adds option to leave python libraries out of the deployment

### PR DESCRIPTION
**Pull request by medwig**
Solves the issue of unnecessarily packaging libraries that are required but included by default by the cloud provider - such as boto3 by AWS. 

Done on a line-by-line basis by adding a #no-deploy comment beside the desired library in requirements.txt. Creates a parsed .requirements.txt file and re-targets pip install to that file.

example:
```
$ cat requirements.txt  
requests    
boto3 #no-deploy
pandas
```

### CODE REVIEW

**Comment by dschep** [comment]
neat suggestion. probably a good enough solution for #14 as well. We're super busy at @UnitedIncome right now, so I'll look at this when I actually have time.

File: index.js [hunk 1]
```diff
@@ -40,12 +40,39 @@ class ServerlessPythonRequirements {
   };
 
   /**
+   * parse requirements.txt into .requirements.txt, leaving out #no-deploy lines
+   * @return true
+   */
+  parseRequirements() {
+    if (!fse.existsSync(path.join(this.serverless.config.servicePath,
+                                  'requirements.txt'))) {
+      return BbPromise.resolve();
```
Comment (dschep) [inline index.js]: Should probably `BbPromise.reject()` since in this scenario the `pip install` command will fail.

File: index.js [hunk 2]
```diff
@@ -40,12 +40,39 @@ class ServerlessPythonRequirements {
...
+    }
+
+    this.serverless.cli.log(
+      `Parsing Python requirements.txt`);
+
+    var fs = require('fs');
```
Comment (dschep) [inline index.js]: just use `fs-extra` which is already imported as `fse`.

File: index.js [hunk 3]
```diff
@@ -40,12 +40,39 @@ class ServerlessPythonRequirements {
...
+    var reqs = fs.readFileSync("requirements.txt").toString().split('\n');
```
Comment (dschep) [inline index.js]: `const` please.

File: index.js [hunk 4]
```diff
@@ -40,12 +40,39 @@ class ServerlessPythonRequirements {
...
+
+    var newReqs = ''
```
Comment (dschep) [inline index.js]: `let` please.

File: index.js [hunk 5]
```diff
@@ -40,12 +40,39 @@ class ServerlessPythonRequirements {
...
+    for (var i in reqs) {
```
Comment (dschep) [inline index.js]: `for (.. of ..)` and template strings:
```
for (const req of reqs) {
  if (req.indexOf('#no-deploy') === -1) {
    newReqs += `${req}\n`;
  }
}
```

File: index.js [hunk 6]
```diff
@@ -40,12 +40,39 @@ class ServerlessPythonRequirements {
...
+      if(reqs[i].indexOf('#no-deploy') == -1) {
+        newReqs += reqs[i] + '\n'
+      }
+    }
+    fs.writeFileSync(".requirements.txt", newReqs, 'utf8');
```
Comment (bsamuel-ui) [inline index.js]: Build tools should use `mktemp` to avoid clobbering existing files whenever possible.

Comment (bsamuel-ui) [inline index.js]: (BTW, just using `mktemp` as an example, you obviously can't write this into `/tmp` and have it be visible.)

**Pull request merged**